File: backend/Models/Review.cs
namespace QBooking.Models
{
    public class Review
    {
        public int Id { get; set; }
        public int BookingId { get; set; }
        public int CustomerId { get; set; }
        public int PropertyId { get; set; }
        public byte OverallRating { get; set; }
        public byte? CleanlinessRating { get; set; }
        public byte? LocationRating { get; set; }
        public byte? ServiceRating { get; set; }
        public byte? ValueRating { get; set; }
        public byte? AmenitiesRating { get; set; }
        public string? Title { get; set; }
        public string? ReviewText { get; set; }
        public string? Pros { get; set; }
        public string? Cons { get; set; }
        public string? TravelType { get; set; }
        public string? RoomStayed { get; set; }
        public bool? IsVerified { get; set; }
        public bool? IsAnonymous { get; set; }
        public string? Status { get; set; }
        public string? HostReply { get; set; }
        public DateTime? HostRepliedAt { get; set; }
        public bool? IsFeatured { get; set; }
        public int? HelpfulCount { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public Booking Booking { get; set; }
        public User Customer { get; set; }
        public Property Property { get; set; }
        public ICollection<ReviewImage> ReviewImages { get; set; } = new List<ReviewImage>();
    }

}

File: frontend/src/types/main/review.ts
// ===== REQUEST TYPES =====

export interface CreateReviewRequest {
  bookingId: number;
  propertyId: number;
  overallRating: number;
  cleanlinessRating?: number;
  locationRating?: number;
  serviceRating?: number;
  valueRating?: number;
  amenitiesRating?: number;
  title?: string;
  reviewText?: string;
  pros?: string;
  cons?: string;
  travelType?: string;
  roomStayed?: string;
  isAnonymous?: boolean;
  images?: File[];
}

export interface UpdateReviewRequest {
  overallRating?: number;
  cleanlinessRating?: number;
  locationRating?: number;
  serviceRating?: number;
  valueRating?: number;
  amenitiesRating?: number;
  title?: string;
  reviewText?: string;
  pros?: string;
  cons?: string;
  travelType?: string;
  roomStayed?: string;
  isAnonymous?: boolean;
  newImages?: File[];
  deleteImageIds?: number[];
}

export interface HostReplyRequest {
  hostReply: string;
}

export interface ReviewFilterRequest {
  page?: number;
  pageSize?: number;
  propertyId?: number;
  minRating?: number;
  maxRating?: number;
  status?: string;
  hasHostReply?: boolean;
  sortBy?: string;
  sortOrder?: string;
}

// ===== RESPONSE TYPES =====

export interface ReviewResponse {
  id: number;
  bookingId: number;
  customerId: number;
  propertyId: number;
  overallRating: number;
  cleanlinessRating?: number;
  locationRating?: number;
  serviceRating?: number;
  valueRating?: number;
  amenitiesRating?: number;
  title?: string;
  reviewText?: string;
  pros?: string;
  cons?: string;
  travelType?: string;
  roomStayed?: string;
  isVerified?: boolean;
  isAnonymous?: boolean;
  status?: string;
  hostReply?: string;
  hostRepliedAt?: string;
  isFeatured?: boolean;
  helpfulCount?: number;
  createdAt?: string;
  updatedAt?: string;
  customer?: CustomerInfo;
  property?: PropertyInfo;
  images: ReviewImageResponse[];
}

export interface ReviewImageResponse {
  id: number;
  imageUrl: string;
  displayOrder?: number;
}

export interface CustomerInfo {
  id: number;
  fullName: string;
  avatar?: string;
}

export interface PropertyInfo {
  id: number;
  name: string;
  slug?: string;
}

export interface ReviewListResponse {
  reviews: ReviewResponse[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}

export interface HostPropertyReviewsResponse {
  propertyId: number;
  propertyName: string;
  totalReviews: number;
  reviewsWithReply: number;
  pendingReviews: number;
  averageRating: number;
}

// ===== API RESPONSE WRAPPER =====

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  statusCode: number;
  data?: T;
  error?: string;
}